configuration/api.go

```diff
@@ -392,3 +396,61 @@ func (a *API) GetLicenseWebhooksState(opts *ManageWebhooksStateOptions) (*Webhoo
 	}, &resp)
 	return resp, err
 }
+
+// DeleteLicenseProperties deletes the properties set within a license.
+func (a *API) DeleteLicenseProperties(props map[string][]string) error {
+	return a.Call("delete_license_properties", &deleteLicensePropertiesRequest{
+		Properties: props,
+	}, &emptyResponse{})
+}
+
+// DeleteGroupProperties deletes the properties set within a group.
+func (a *API) DeleteGroupProperties(id int, props map[string][]string) error {
+	return a.Call("delete_group_properties", &deleteGroupPropertiesRequest{
+		ID:         id,
+		Properties: props,
+	}, &emptyResponse{})
+}
+
+// AddAutoAccess creates an auto access data structure.
+func (a *API) AddAutoAccess(groupIDs []int, url, domain *Condition, geolocation *GeolocationCondition, desc, nextID string) (string, error) {
+	var resp addAutoAccessResponse
+	req := addAutoAccessRequest{
+		Description: desc,
+		NextID:      nextID,
+	}
+	req.Access.Groups = groupIDs
+	req.Conditions.Url = url
+	req.Conditions.Domain = domain
+	req.Conditions.Geolocation = geolocation
+	err := a.Call("add_auto_access", &req, &resp)
+
+	return resp.ID, err
+}
+
+// UpdateAutoAccess updates an existing auto access.
+func (a *API) UpdateAutoAccess(id string, groupIDs []int, url, domain *Condition, geolocation *GeolocationCondition, desc, nextID string) error {
+	req := updateAutoAccessRequest{
+		addAutoAccessRequest: addAutoAccessRequest{
```

Comment:
This looks weird, I think it'd be better to extract common `autoAccessRequest` struct (without `add`)

Reply:
I think i remove weird part, is it better now?